integration_tests/test_full_workflow.py

```diff
@@ -47,3 +47,75 @@ def test_emscoring_workflow(caplog, monkeypatch):
         assert os.path.isdir("analysis/1_emscoring_analysis") is True
         # there should be a report.html inside it
         assert os.path.isfile("analysis/1_emscoring_analysis/report.html") is True
+
+
+def test_interactive_analysis_on_workflow(monkeypatch):
+    """A comprehensive test for the interactive commands and analysis."""
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # copy the files to the temporary directory
+        files = [
+            "cyclic-peptide.pdb",
+            "prot.pdb",
+            "workflow.cfg",
+        ]
+        for fl in files:
+            shutil.copy(
+                Path(GOLDEN_DATA, fl),
+                Path(tmpdir, fl),
+            )
+
+        monkeypatch.chdir(tmpdir)
+
+        from haddock.clis.cli import main as cli_main
+        cli_main(
+            Path("workflow.cfg"),
+        )
+        # read run_dir in workflow.cfg
+        with open("workflow.cfg", "r") as f:
+            for line in f:
+                if "run_dir" in line:
+                    run_dir = line.split("=")[1].strip().strip('"')
+                    break
+        assert os.path.isdir(run_dir) is True
+        assert os.path.isdir(Path(run_dir, "0_topoaa")) is True
+        assert os.path.isdir(Path(run_dir, "1_rigidbody")) is True
+        assert os.path.isdir(Path(run_dir, "2_clustfcc")) is True
+        assert os.path.isdir(Path(run_dir, "3_caprieval")) is True
+        assert os.path.isdir(Path(run_dir, "analysis")) is True
+
+        # now running interactive re-clustering
+        clustfcc_dir = f"{run_dir}/2_clustfcc"
+        from haddock.clis.cli_re import maincli
+        monkeypatch.setattr("sys.argv",
+                            ["haddock3-re", "clustfcc", clustfcc_dir, "-f", "0.7"]
+                            )
+        maincli()
```

Comment:
If I understand well what's happening here:
- faking a `sys.argv` input for the `haddock3-re` CLI
- running the CLI `maincli()`, that will use `sys.argv` information through the `argparse` library
- doing its stuff.

Why not importing the `from haddock.clis.re.clustfcc import reclustfcc` and running it ?

I think a little comment explaining what's happening here could help future people who will maintain this part of the code.

Reply:
I really wanted to call the main haddock3-re cli to mimic a real-case scenario (such as Stefan's one).

will add a comment there